src/SPC/builder/macos/MacOSBuilder.php

```diff
@@ -300,13 +306,15 @@ protected function buildEmbed(): void
             ->exec('rm ' . BUILD_ROOT_PATH . '/lib/libphp.a')
             ->exec('ar rcs ' . BUILD_ROOT_PATH . '/lib/libphp.a *.o')
             ->exec('rm -Rf ' . BUILD_ROOT_PATH . '/lib/php-o');
-        FileSystem::replaceFileStr(BUILD_BIN_PATH . '/phpize', "prefix=''", "prefix='" . BUILD_ROOT_PATH . "'");
-        FileSystem::replaceFileStr(BUILD_BIN_PATH . '/phpize', 's##', 's#/usr/local#');
-        $php_config_str = FileSystem::readFile(BUILD_BIN_PATH . '/php-config');
-        str_replace('prefix=""', 'prefix="' . BUILD_ROOT_PATH . '"', $php_config_str);
-        // move mimalloc to the beginning of libs
-        $php_config_str = preg_replace('/(libs=")(.*?)\s*(' . preg_quote(BUILD_LIB_PATH, '/') . '\/mimalloc\.o)\s*(.*?)"/', '$1$3 $2 $4"', $php_config_str);
-        FileSystem::writeFile(BUILD_BIN_PATH . '/php-config', $php_config_str);
+        $this->patchPhpScripts();
+    }
+
+    protected function buildPhpDev(): void
+    {
+        $vars = SystemUtil::makeEnvVarString($this->getMakeExtraVars());
+        shell()->cd(SOURCE_PATH . '/php-src')
+            ->exec(getenv('SPC_CMD_PREFIX_PHP_MAKE') . ' INSTALL_ROOT=' . BUILD_ROOT_PATH . " {$vars} install-build install-programs install-headers");
+        $this->patchPhpScripts();
```

Comment:
like I said below, get rid of this duplication

Reply:
Do you think it is better to have an error message or implicitly add embed SAPI support when specifying `--with-dynamic` but not `--build-embed`?

Reply:
`--with-shared` instead of `--with-dynamic` and I think it's fine to implicitly enable --build-embed.

Reply:
I've been wondering if there's a more appropriate parameter name; using `--with-shared` or `--with-dynamic` is too vague, but `--with-shared-extensions` is a bit long.

Reply:
Thinking about it, to preserve future compatibility, I think we should call it `--build-shared`
For the simple reason that `--with-shared="zip"` could also mean we're building libphp.a/libphp.so which will dynamically load zip.